src/components/screens/editor/editor-screen.tsx

```diff
@@ -8,64 +8,59 @@ import Api from "api";
 import { NavigationTabContainer } from "styled/layouts/navigations";
 import NavigationTab from "components/layouts/navigations/navigation-tab";
 import { Metaform, MetaformVersion, MetaformVersionType } from "generated/client";
-import { AdminFormListStack, AdminFormTypographyField, VersionListHeader } from "styled/react-components/react-components";
-import { DataGrid, GridColDef } from "@mui/x-data-grid";
 import { NewUserButton } from "styled/layouts/admin-layout";
-import moment from "moment";
-import EditorScreenDrawer from "./editor-screen-drawer";
+import EditorScreenDrawer from "../../editor/editor-screen-drawer";
 import { useNavigate } from "react-router-dom";
-
-/**
- * Interface for single Metaform row
- */
-interface Row {
-  metaform: Metaform;
-  versions: MetaformVersion[];
-}
+import GenericLoaderWrapper from "components/generic/generic-loader";
+import EditorScreenTable from "../../editor/editor-screen-table";
 
 /**
  * Editor screen component
  */
 const EditorScreen: React.FC = () => {
+  const currentPath = window.location.pathname;
   const errorContext = useContext(ErrorContext);
   const navigate = useNavigate();
 
   const apiClient = useApiClient(Api.getApiClient);
   const { metaformsApi, versionsApi } = apiClient;
 
-  const [ metaformRows, setMetaformRows ] = useState<Row[]>([]);
+  const [ metaforms, setMetaforms ] = useState<Metaform[]>([]);
+  const [ metaformVersions, setMetaformVersions ] = useState<MetaformVersion[]>([]);
+
   const [ loading, setLoading ] = useState<boolean>(false);
 
   const [ drawerOpen, setDrawerOpen ] = useState<boolean>(false);
 
+  /* eslint-disable @typescript-eslint/return-await */
   /**
-   * Gets Metaforms and corresponding MetaformVersions 
+   * Gets Metaforms and MetaformVersions 
    */
   const loadMetaforms = async () => {
     setLoading(true);
     
     try {
       const forms = await metaformsApi.listMetaforms({});
-      const rows: Row[] = await Promise.all(forms.map(async form => {
-        return {
-          metaform: form,
-          versions: await versionsApi.listMetaformVersions({ metaformId: form.id! })
-        };
-      }));
+      const versions = await Promise.all(forms.map(form => versionsApi.listMetaformVersions({ metaformId: form.id! })));
 
-      setMetaformRows(rows);
+      setMetaforms(forms);
+      setMetaformVersions(versions.flat());
     } catch (e) {
       errorContext.setError(strings.errorHandling.adminFormsScreen.listForms, e);
     }
 
     setLoading(false);
   };
+  /* eslint-enable @typescript-eslint/return-await */
 
   /**
-   * Creates new Metaform and MetaformVersion
+   * Creates new Metaform and  MetaformVersion and navigates to DraftEditorScreen
    */
   const createMetaform = async (metaform: Metaform) => {
     try {
+      if (metaforms.find(existingForm => existingForm.title === metaform.title)) {
+        throw new Error(strings.editorScreen.createFormDuplicateNameError);
```

Comment:
Move string to errorHandling (strings.errorHandling.editorScreen.createFo...)

Reply:
Done